sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/NioBlobInputStream.java

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.nio;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.specialized.BlobInputStream;
+import com.azure.storage.common.implementation.Constants;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Provides an InputStream to read a file stored as an Azure Blob.
+ */
+public class NioBlobInputStream extends InputStream {
```

Comment:
Are you wrapping each method since you want to log using the NioBlobInputStream class? Just wondering

Reply:
+1 I wonder what additional functionality `NioBlobInputStream` provides over `BlobInputStream`.

Reply:
There's no additional functionality right now, but I can't say for sure that usage patterns will be the same between nio and the sdk, and they may require swapping out the implementation for different performance characteristics or adding features we don't want in the sdk. Because of that, I think it's safer to define a type specific to nio that we can update independent of the underlying BlobInputStream

Reply:
Hmm. Then the only reason this exist would be to reserve publicly available class name (+/- error translation depending on other discussion). Otherwise this could be [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) .